src/dto/User/create.user.dto.ts

```diff
@@ -0,0 +1,45 @@
+import Joi from 'joi';
+
+const generatePassword = () => {
+    return new RegExp('^[a-zA-Z0-9]{3,30}$')
+};
+
+export const CreateUserSchema = Joi.object({
+    firstname: Joi.string()
+        .alphanum()
+        .min(3)
+        .max(30)
+        .required(),
```

Comment:
You could create reusable thunk from that. Smth like this:
```ts
const firstnameAndLastnameValidation = Joi.string()
        .alphanum()
        .min(3)
        .max(30)
        .required()
```
And use it here and in lastname.

Reply:
+1